core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -137,11 +137,7 @@ public void write(T row) throws IOException {
       StructLike copiedKey = StructCopy.copy(structProjection.wrap(asStructLike(row)));
 
       // Adding a pos-delete to replace the old path-offset.
-      PathOffset previous = insertedRowMap.put(copiedKey, pathOffset);
-      if (previous != null) {
-        // TODO attach the previous row if has a positional-delete row schema in appender factory.
-        posDeleteWriter.delete(previous.path, previous.rowOffset, null);
-      }
```

Comment:
Removing this doesn't seem like a good idea.
Shouldn't we have to write a position delete row if a previously inserted row is updated?

Reply:
Thans for your review!

Reply:
> Removing this doesn't seem like a good idea. Shouldn't we have to write a position delete row if a previously inserted row is updated?

Thanks for your review!
previous is always null when upsert is true, as BaseEqualityDeltaWriter.deleteKey() is called when upsert is true. Otherwise, when write.upsert.enabled is false, I think it should not delete previous row.  Here I